apps/browser-extension-wallet/src/hooks/useWalletManager.ts

```diff
@@ -391,6 +392,31 @@ export const useWalletManager = (): UseWalletManager => {
     return (storedChain?.chainName && chainIdFromName(storedChain.chainName)) || DEFAULT_CHAIN_ID;
   }, [currentChain]);
 
+  const getWalletInfo = useCallback(async (): Promise<Wallet.WalletDisplayInfo | undefined> => {
+    const { activeBlockchain } = await backgroundService.getBackgroundStorage();
+
+    const [activeWallet, bitcoinActiveWallet] = await firstValueFrom(
+      combineLatest([walletManager.activeWalletId$, bitcoinWalletManager.activeWalletId$])
+    );
+
+    const walletId = activeBlockchain === 'bitcoin' ? bitcoinActiveWallet?.walletId : activeWallet?.walletId;
+    const wallets = await firstValueFrom(walletRepository.wallets$);
+    const wallet = wallets.find((w) => w.walletId === walletId);
+
+    let accountMetadata;
+    if (wallet && isAnyBip32Wallet(wallet)) {
+      const account = wallet.accounts.find((a) => a.accountIndex === wallet.metadata.lastActiveAccountIndex);
```

Comment:
since wallet metadata is not updated in the `addAccount` flow https://github.com/input-output-hk/lace/blob/main/apps/browser-extension-wallet/src/hooks/useWalletManager.ts#L1366, `lastActiveAccountIndex` won't be accurate here

```suggestion
      const account = wallet.accounts.find((a) => a.accountIndex === cardanoWallet?.source.account?.accountIndex);
```
or please align the implementation of `addAccount` with `activateWallet` so it also updates the wallet metadata

Reply:
resolved